src/java.base/share/classes/java/util/zip/ZipFile.java

```diff
@@ -1307,6 +1317,15 @@ private void checkZip64ExtraFieldValues(int off, int blockSize, long csize,
             if (!isZip64ExtBlockSizeValid(blockSize)) {
                 zerror("Invalid CEN header (invalid zip64 extra data field size)");
             }
+            // if ZIP64_EXTID blocksize == 0, validate csize and size
```

Comment:
If you put this block in front of the call to `isZip64ExtBlockSizeValid()` we don't have to handle the `blockSize == 0` case in `isZip64ExtBlockSizeValid()`.

This will also make the following comment true again:
```
            // Note we do not need to check blockSize is >= 8 as
            // we know its length is at least 8 from the call to
            // isZip64ExtBlockSizeValid()
```